lib/db/mem.js

```diff
@@ -1073,6 +1074,44 @@ module.exports = function (log, error) {
     return P.resolve({})
   }
 
+  Memory.prototype.createSigninCode = (code, uid, createdAt) => {
+    code = code.toString('hex')
+
+    if (signinCodes[code]) {
+      return P.reject(error.duplicate())
+    }
+
+    signinCodes[code] = {
+      uid: uid.toString('hex'),
+      createdAt
+    }
+
+    return P.resolve({})
+  }
+
+  Memory.prototype.useSigninCode = code => {
+    code = code.toString('hex')
+
+    if (! signinCodes[code]) {
+      return P.reject(error.notFound())
+    }
+
+    const email = accounts[signinCodes[code].uid.toString('hex')].email
+    delete signinCodes[code]
+
+    return P.resolve({ email })
```

Comment:
If/when we implement phase 4 of this feature, then there'll be more stuff returned yep. But let's do that when it's time I reckon, who's to say how the requirements may change for that phase in the meantime?